packages/gatsby/src/bootstrap/requires-writer.ts

```diff
@@ -184,11 +186,27 @@ export const writeAll = async (state: IGatsbyState): Promise<boolean> => {
   const pages = [...state.pages.values()]
   const matchPaths = getMatchPaths(pages)
   const components = getComponents(pages)
+  let cleanedSSRVisitedPageComponents: Array<IGatsbyPageComponent> = components
```

Comment:
Is there a reason to assign `components` here? We later pass `cleanedSSRVisitedPageComponents` to `createHash` below (line 233) and so this affects the current stable codepath - without the flag we pass all components to `createHash`.

Probably not critical but seems unnecessary.

Reply:
This was the simplest way I could see to make typescript happy 🤷 What's the better way to assign a type here?

Reply:
I think empty array should work?